crates/phactory/src/prpc_service.rs

```diff
@@ -1127,8 +1138,9 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
         _request: (),
     ) -> RpcResult<pb::HandoverChallenge> {
         let mut phactory = self.lock_phactory();
+        let (block, ts) = phactory.current_block()?;
```

Comment:
Add comments here about why not use `system.now_ms` and `system.block_number`?

Reply:
Added alert comments on the fields themselves.